x86_64-pc-windows-gnu: ld: cannot find -lcredui and -lsecur32 after upgrading to rustc 1.26.0-nightly (521d91c6b 2018-03-14)

https://ci.appveyor.com/project/laumann/compiletest-rs/build/1.0.38/job/o8rkn779qh7ios49

> error: linking with `gcc` failed: exit code: 1
  |
  = note: "gcc" "-Wl,--enable-long-section-names" "-fno-use-linker-plugin" "-Wl,--nxcompat" "-nostdlib" "-m64" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\dllcrt2.o" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\rsbegin.o" "-L" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive0.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive1.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive10.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive11.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive12.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive13.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive14.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive15.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive2.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive3.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive4.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive5.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive6.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive7.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive8.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive9.rcgu.o" "-o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.dll" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.crate.metadata.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.crate.allocator.rcgu.o" "-Wl,--gc-sections" "-nodefaultlibs" "-L" "C:\\projects\\compiletest-rs\\target\\debug\\deps" "-L" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-Wl,-Bstatic" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\libserde_derive_internals-555f0e717120011b.rlib" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\libsyn-9d73a1b0d8b88639.rlib" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\libquote-cce34627b501f031.rlib" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\libproc_macro2-73d79182fd654d90.rlib" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\libunicode_xid-25214f6da763bedd.rlib" "-L" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-Wl,-Bdynamic" "-l" "proc_macro-93dbabd980fbabf9" "-L" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-l" "syntax-031c8126770b77bb" "-L" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-l" "rustc_errors-f2eda81f3e9dcbb7" "-L" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-l" "syntax_pos-f46872503870f814" "-L" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-l" "rustc_data_structures-c0d56ce5645d455f" "-L" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-l" "serialize-60f2dac169f8734f" "-L" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-l" "rustc_cratesio_shim-ce4463286aac2a73" "-L" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-l" "std-6b18c18bbc608054" "-Wl,-Bstatic" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libcompiler_builtins-362c10b9d34fdd8e.rlib" "-Wl,-Bdynamic" "-l" "kernel32" "-l" "credui" "-l" "setupapi" "-l" "user32" "-l" "secur32" "-l" "advapi32" "-l" "winspool" "-l" "msimg32" "-l" "gdi32" "-l" "opengl32" "-l" "kernel32" "-l" "dbghelp" "-l" "advapi32" "-l" "ws2_32" "-l" "userenv" "-l" "shell32" "-shared" "-lmingwex" "-lmingw32" "-lgcc" "-lmsvcrt" "-lmsvcrt" "-luser32" "-lkernel32" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\rsend.o"
  = note: ld: cannot find -lcredui
          ld: cannot find -lsecur32
>          
> error: aborting due to previous error
> error: Could not compile `serde_derive`.


It builds fine with nightly-2018-03-07

https://ci.appveyor.com/project/laumann/compiletest-rs/build/1.0.38/job/rcfcfv3mfmx7h6t3

The relevant code here: https://github.com/laumann/compiletest-rs/pull/108

Oh no, rustup-toolchain-install-master does not build.

> error: failed to run custom build command for `lzma-sys v0.1.9`
process didn't exit successfully: `C:\Users\appveyor\AppData\Local\Temp\1\cargo-install.awHWa8SE4m3J\release\build\lzma-sys-41f0afb77098a370\build-script-build` (exit code: 101)
--- stdout
cargo:rerun-if-env-changed=LZMA_API_STATIC
cargo:rustc-link-search=C:\Users\appveyor\AppData\Local\Temp\1\cargo-install.awHWa8SE4m3J\release\build\lzma-sys-19ff7df3ceadee56\out/lib
cargo:root=C:\Users\appveyor\AppData\Local\Temp\1\cargo-install.awHWa8SE4m3J\release\build\lzma-sys-19ff7df3ceadee56\out
cargo:include=C:\Users\appveyor\AppData\Local\Temp\1\cargo-install.awHWa8SE4m3J\release\build\lzma-sys-19ff7df3ceadee56\out/include
cargo:rerun-if-changed=xz-5.2.3/configure
cargo:rustc-link-lib=static=lzma
OPT_LEVEL = Some("3")
TARGET = Some("x86_64-pc-windows-gnu")
HOST = Some("x86_64-pc-windows-gnu")
TARGET = Some("x86_64-pc-windows-gnu")
TARGET = Some("x86_64-pc-windows-gnu")
HOST = Some("x86_64-pc-windows-gnu")
CC_x86_64-pc-windows-gnu = None
CC_x86_64_pc_windows_gnu = None
HOST_CC = None
CC = None
TARGET = Some("x86_64-pc-windows-gnu")
HOST = Some("x86_64-pc-windows-gnu")
CFLAGS_x86_64-pc-windows-gnu = None
CFLAGS_x86_64_pc_windows_gnu = None
HOST_CFLAGS = None
CFLAGS = None
DEBUG = Some("false")
running: "sh" "/C/Users/appveyor/AppData/Local/Temp/1/cargo-install.awHWa8SE4m3J/release/build/lzma-sys-19ff7df3ceadee56/out/src/configure" "--prefix=/C/Users/appveyor/AppData/Local/Temp/1/cargo-install.awHWa8SE4m3J/release/build/lzma-sys-19ff7df3ceadee56/out" "--disable-doc" "--disable-lzma-links" "--disable-lzmainfo" "--disable-lzmadec" "--disable-xz" "--disable-xzdec" "--disable-scripts" "--disable-shared" "--disable-nls" "--disable-rpath" "--enable-threads=win95"
XZ Utils 5.2.3
System type:
checking build system type... x86_64-pc-msys
checking host system type... x86_64-pc-msys
Configure options:
checking if debugging code should be compiled... no
checking which encoders to build... lzma1 lzma2 delta x86 powerpc ia64 arm armthumb sparc
checking which decoders to build... lzma1 lzma2 delta x86 powerpc ia64 arm armthumb sparc
checking which match finders to build... hc3 hc4 bt2 bt3 bt4
checking which integrity checks to build... crc32 crc64 sha256
checking if external SHA-256 should be used... no
checking if assembler optimizations should be used... x86_64
checking if small size is preferred over speed... no
checking if threading support is wanted... yes, win95
checking how much RAM to assume if the real amount is unknown... 128 MiB
checking if library symbol versioning should be used... no
checking if sandboxing should be used... no
checking for a shell that conforms to POSIX... /bin/sh
Initializing Automake:
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... no
checking whether make supports nested variables... no
checking whether ln -s works... no, using cp -pR
checking for style of include used by make... none
checking for gcc... gcc.exe
checking whether the C compiler works... no
--- stderr
configure: error: in `/tmp/cargo-install.awHWa8SE4m3J/release/build/lzma-sys-19ff7df3ceadee56/out/build':
configure: error: C compiler cannot create executables
See `config.log' for more details
thread 'main' panicked at 'assertion failed: try_run(cmd)', C:\Users\appveyor\.cargo\registry\src\github.com-1ecc6299db9ec823\lzma-sys-0.1.9\build.rs:182:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.

--- a/src/bootstrap/dist.rs
+++ b/src/bootstrap/dist.rs
@@ -203,11 +203,14 @@ fn make_win_dist(
         "libbcrypt.a",
         "libcomctl32.a",
         "libcomdlg32.a",
+        "libcredui.a",
         "libcrypt32.a",
+        "libdbghelp.a",
         "libgdi32.a",
         "libimagehlp.a",
         "libiphlpapi.a",
         "libkernel32.a",
+        "libmsimg32.a",
         "libmsvcrt.a",
         "libodbc32.a",
         "libole32.a",
@@ -215,6 +218,7 @@ fn make_win_dist(
         "libopengl32.a",
         "libpsapi.a",
         "librpcrt4.a",
+        "libsecur32.a",
         "libsetupapi.a",
         "libshell32.a",
         "libuser32.a",
@@ -225,8 +229,6 @@ fn make_win_dist(
         "libwinspool.a",
         "libws2_32.a",
         "libwsock32.a",
-        "libdbghelp.a",
-        "libmsimg32.a",
     ];
 
     //Find mingw artifacts we want to bundle

